### ISSUE TITLE
Unable to initialize phoenix absinthe socket connection

### ISSUE THREAD

**Issue by akeating** [dillonkearns/graphqelm#30]
I have been comparing the messages sent from the phoenix javascript lib and this lib. The init message fails (doesn't match) deep in phoenix..
```
(FunctionClauseError) no function clause matching in Phoenix.Socket.Message.from_map!/1
            (phoenix) lib/phoenix/socket/message.ex:22: Phoenix.Socket.Message.from_map!([nil, "1", "__absinthe__:control", "phx_join", %{}]
```

I'm thinking it's because this line:
https://github.com/dillonkearns/graphqelm/blob/c240c755f62f9ecaa7b68ae11820da0048048239/src/Graphqelm/Subscription/Protocol.elm#L89
encodes null instead of "1". I notice your code comments indicate the "1" in that position too.

Your subscriptions support looks promising so far, although as an Elm noob, I find the abstractions a little hard to follow.

**Comment by dillonkearns** [comment]
Hello Adam, thank you for reporting this! So I am able to connect with `Graphqelm.Subscription.Protocol.phoenixAbsinthe` with the following setup:

* Absinthe/Phoenix code, running at https://graphqelm.herokuapp.com: https://github.com/dillonkearns/graphqelm-demo
* This chat example code (see it [on Ellie here](https://rebrand.ly/graphqelm-subscriptions)): https://github.com/dillonkearns/graphqelm/blob/master/examples/src/Subscription.elm

So it looks like I'm on Phoenix 1.3.0, Absinthe-Phoenix 1.4.0, Absinthe 1.4.5.

Here is what my websocket frames look like in the dev tools:
<img width="885" alt="__src_subscription_elm" src="https://user-images.githubusercontent.com/1384166/35575311-e72cd00a-0590-11e8-9fbd-3a4482e8ec8a.png">

Could you provide the following information to help debug this? It would be really helpful for me to get better insight into how to support different Phoenix versions:

* Your Phoenix version
* A screenshot of your websocket frames from the dev tools window (like mine above)
* Try changing it to a "1" instead of null in that spot and let me know if that fixes it.
You can build your own Protocol as pass it in to `Subscription.init` (just `import Graphqelm.Subscription.Protocol exposing (Protocol)` and copy paste [this code](https://github.com/dillonkearns/graphqelm/blob/c240c755f62f9ecaa7b68ae11820da0048048239/src/Graphqelm/Subscription/Protocol.elm#L84-L120) into your file).

Thanks for filing the issue, it's super helpful for working out the kinks on various framework subtleties! And let me know if you have questions on how to use Graphqelm Subscriptions, I'd be happy to help you out on the Elm Slack.

**Comment by akeating** [comment]
Thanks for the detailed reply. So I just spent more time on this:
1. I realized your example socket url has a query param of `vsn=2.0.0` on it and it looks like that is required for an initial response.
2. The `null` or `"1"` in the first position appears to have no obvious effect.

I now have elm <-> phoenix absinthe <-> vuejs subscriptions working. Nice job!
